src/main/scala/ch/epfl/bluebrain/nexus/kg/indexing/ElasticIndexer.scala

```diff
@@ -65,14 +68,21 @@ class ElasticIndexer[F[_]](client: ElasticClient[F], index: String, resources: R
     client
       .get[Json](index, config.docType, id.elasticId, include = Set(revKey))
       .map(j => j.hcursor.get[Long](revKey).toOption)
+      .handleError {
+        case ElasticClientError(StatusCodes.NotFound, _) => None
+      }
 
   private def indexResource(res: Resource): F[Unit] = {
     val primaryNode = Some(IriNode(res.id.value))
     val graph       = res.metadata ++ res.typeGraph
 
     val payload = graph.asJson(resourceCtx, primaryNode).getOrElse(graph.asJson)
-    val merged  = payload deepMerge Json.obj("_source" -> Json.fromString(res.value.noSpaces))
-    client.update(index, config.docType, res.id.elasticId, merged)
+    val merged  = payload deepMerge Json.obj("_original_source" -> Json.fromString(res.value.noSpaces))
+    val esPayload = Json.obj(
+      "doc"           -> merged,
+      "doc_as_upsert" -> Json.fromBoolean(true)
+    )
+    client.update(index, config.docType, res.id.elasticId, esPayload)
```

Comment:
Am I missing something or we are doing an update, while we should be doing a full replace?
From the docs:
> The update API also support passing a partial document, which will be merged into the existing document (simple recursive merge, inner merging of objects, replacing core "keys/values" and arrays). To fully replace the existing document, the index API should be used instead. The following partial update adds a new field to the existing document

Reply:
You're right, we should be doing what this [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/_modifying_your_data.html) says, but instead we're doing an [_update](https://www.elastic.co/guide/en/elasticsearch/reference/current/_updating_documents.html). I'll fix that.

Reply:
Yes, it is a regular [create call](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-index_.html)